samples/browser/src/geckoNightly/java/org/mozilla/samples/browser/Components.kt

```diff
@@ -5,16 +5,35 @@ package org.mozilla.samples.browser
 
 import android.content.Context
 import mozilla.components.browser.engine.gecko.GeckoEngine
+import mozilla.components.browser.engine.gecko.glean.GeckoAdapter
 import mozilla.components.concept.engine.Engine
 import mozilla.components.feature.webcompat.WebCompatFeature
 import mozilla.components.support.base.log.Log
+import org.mozilla.geckoview.GeckoRuntime
+import org.mozilla.geckoview.GeckoRuntimeSettings
 
 /**
  * Helper class for lazily instantiating components needed by the application.
  */
 class Components(private val applicationContext: Context) : DefaultComponents(applicationContext) {
+    private var runtime: GeckoRuntime? = null
+
+    @Synchronized
+    private fun getOrCreateRuntime(context: Context): GeckoRuntime {
+        if (runtime == null) {
+            val builder = GeckoRuntimeSettings.Builder()
+
+            // Allow for exfiltrating Gecko metrics through the Glean SDK.
+            builder.telemetryDelegate(GeckoAdapter())
+
+            runtime = GeckoRuntime.create(context, builder.build())
+        }
+
+        return runtime!!
+    }
+
     override val engine: Engine by lazy {
-        GeckoEngine(applicationContext, engineSettings).also {
+        GeckoEngine(applicationContext, engineSettings, getOrCreateRuntime(applicationContext)).also {
```

Comment:
nit: RB has this more complex setup with `getOrCreateRuntime` since we need the runtime for other things too in RB. Here you could simplify it and get rid of the method and property and just create the runtime in the lazy block of `Engine` directly (since that gets executed only once already).

Reply:
Oh, thanks. Fixed this :)